Makefile

```diff
@@ -13,6 +13,8 @@ build:
 	coqc compare_regex.v
 	coqc reduce_orb.v
 
+	coqc derive_def.v
```

Comment:
Can't decide which file should have extra naming.
derive_def or derive_theorems, either way I am unhappy :)
But this is exactly what I was asking for.

Reply:
Please ignore this rambling.

Reply:
I agree with you - I'm not happy with this. But what makes it slightly better is that because `derive.v` imports `derive_def.v` using `Requires Export` then any module that imports `derive.v` will automatically get the symbols from `derive_def.v` in scope.

So the only modules that should import `derive_def.v` are `setoid.v` and `derive.v`.

Reply:
Oh that is very cool @ `Requires Export` maybe also worth a comment?